src/libraries/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -659,4 +659,16 @@
   <data name="JsonPolymorphismOptionsAssociatedWithDifferentJsonTypeInfo" xml:space="preserve">
     <value>Parameter already associated with a different JsonTypeInfo instance.</value>
   </data>
+  <data name="JsonPropertyRequiredAndNotDeserializable" xml:space="preserve">
+    <value>JsonPropertyInfo with name '{0}' for type '{1}' is required but is not deserializable.</value>
+  </data>
+  <data name="JsonPropertyRequiredAndExtensionData" xml:space="preserve">
+    <value>JsonPropertyInfo with name '{0}' for type '{1}' cannot be required and be an extension data property.</value>
+  </data>
+  <data name="JsonRequiredPropertiesMissingHeader" xml:space="preserve">
+    <value>Type '{0}' contains required properties which were not set during deserialization:</value>
+  </data>
+  <data name="JsonRequiredPropertiesMissingItem" xml:space="preserve">
+    <value>- {0}</value>
```

Comment:
This looks accidental?

Reply:
this is intentional. the error prints as:

```
Type 'Foo' contains required properties which were not set during deserialization:
- BarProperty
- BazProperty
```

I split those in two in the resources

Reply:
Do we have precedent in doing this elsewhere? This feels to me like something we might as well hardcode in the message builder.

Reply:
I'm not sure how would that localize in other languages (i.e. are hyphens always on the left in all languages? is hyphen meaningful in all languages?) The only precedence in JSON I found is:
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/src/Resources/Strings.resx#L454-L456

but that uses always two element list which is easier to translate I guess. I'll see if I can find precedence, if I don't find I will file an issue to have someone double check

Reply:
Not sure what best practice might be here. Maybe @carlossanlop or @ericstj might know.

On an unrelated note, I don't think the exception message should enumerate _every_ missed required property. Consider what exception message might materialize when you deserialize `{}` in a POCO with 1000 required properties. It might be best to pick a good cut-off number like 3 or 4.

Reply:
The closest I found to us producing a list is: https://github.com/dotnet/runtime/blob/e55c908229e36f99a52745d4ee85316a0e8bb6a2/src/libraries/System.ComponentModel.TypeConverter/src/System/Drawing/PointConverter.cs#L71

so basically `culture.TextInfo.ListSeparator` + space. I'll follow that pattern rather than new line